### PR TITLE
add async

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/UserService.Broker/Consumers/AccessValidatorConsumer.cs [hunk 1]
```diff
@@ -8,25 +8,25 @@
...
-            await context.RespondAsync<IOperationResult<bool>>(response);
-        }
+      await context.RespondAsync<IOperationResult<bool>>(response);
+    }
 
-        private object IsAdmin(Guid userId)
-        {
-            return repository.Get(userId).IsAdmin;
-        }
+    private async Task<object> IsAdminAsync(Guid userId)
```
Comment (BelehovEgor) [inline src/UserService.Broker/Consumers/AccessValidatorConsumer.cs:27]: move up to public methods

File: src/UserService.Broker/Consumers/AccessValidatorConsumer.cs [hunk 2]
```diff
@@ -8,25 +8,25 @@
 
 namespace LT.DigitalOffice.UserService.Broker.Consumers
 {
-    public class AccessValidatorConsumer : IConsumer<ICheckUserIsAdminRequest>
-    {
-        private readonly IUserRepository repository;
+  public class AccessValidatorConsumer : IConsumer<ICheckUserIsAdminRequest>
+  {
+    private readonly IUserRepository repository;
```
Comment (BelehovEgor) [inline src/UserService.Broker/Consumers/AccessValidatorConsumer.cs:13]: rename to _repository

File: src/UserService.Broker/Consumers/CheckUsersExistenceConsumer.cs [hunk 3]
```diff
@@ -8,27 +8,27 @@
...
 
-            await context.RespondAsync<IOperationResult<ICheckUsersExistence>>(response);
-        }
+      await context.RespondAsync<IOperationResult<ICheckUsersExistence>>(response);
+    }
 
-        private object GetUsersExistenceInfo(ICheckUsersExistence requestIds)
-        {
-            List<Guid> userIds = _repository.AreExistingIds(requestIds.UserIds);
+    private async Task<object> GetUsersExistenceInfoAsync(ICheckUsersExistence requestIds)
```
Comment (BelehovEgor) [inline src/UserService.Broker/Consumers/CheckUsersExistenceConsumer.cs:27]: move up to public methods

File: src/UserService.Business/Commands/Password/ForgotPasswordCommand.cs [hunk 4]
```diff
@@ -20,131 +18,136 @@
...
+      string email,
+      Guid secret,
+      List<string> errors)
+    {
+      //TODO: fix add specific template language
+      string templateLanguage = "en";
+      EmailTemplateType templateType = EmailTemplateType.Warning;
+      try
+      {
+        var templateValues = ISendEmailRequest.CreateTemplateValuesDictionary(
```
Comment (BelehovEgor) [inline src/UserService.Business/Commands/Password/ForgotPasswordCommand.cs:61]: use explicit type

File: src/UserService.Validation/Education/CreateEducationRequestValidator.cs [hunk 5]
```diff
@@ -5,25 +5,23 @@
...
+  public class CreateEducationRequestValidator : AbstractValidator<CreateEducationRequest>, ICreateEducationRequestValidator
+  {
+    public CreateEducationRequestValidator(IUserRepository _userRepository)
     {
-        public CreateEducationRequestValidator(IUserRepository repository)
-        {
-            RuleFor(education => education.UserId)
-                .Must(id => repository.IsUserExist(id));
+      RuleFor(education => education.UserId)
+        .MustAsync(async (e, token, context) => await _userRepository.IsUserExistAsync(e.UserId));
```
Comment (BelehovEgor) [inline src/UserService.Validation/Education/CreateEducationRequestValidator.cs:13]: you can write (e, _) or (e, _, _ ) if you don't need other variable

**Review by BelehovEgor — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Pull request merged**